### PR TITLE
Run pytest suite in CI

### LINKED ISSUE #3: Add CI Integration Tests
Should pull and spin up a mock AllSpice Hub instance to ensure API functions don't regress.

### PR DESCRIPTION
**Pull request by shrik450**
Closes #3.

Adds a GitHub actions workflow to run our pytest suite in CI. A few changes to the tests were required to make the test suite work with a fresh instance of AllSpice Hub.

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -137,11 +139,29 @@ def test_get_repository(instance):
...
 
+def test_add_content_to_repo(instance):
+    repo = Repository.request(instance, test_org, test_repo)
+    file_content = open("tests/data/test.pcbdoc", "rb").read()
+    file_content = base64.b64encode(file_content).decode("utf-8")
+    repo.create_file("test.pcbdoc", file_content)
+    assert len(repo.get_commits()) == 2
+    assert [content.name for content in repo.get_git_content()] == ["README.md", "test.pcbdoc"]
+
+    # As an optimization, we call the generated json and svg methods so we kick off
```
Comment (shrik450) [inline tests/test_api.py:150]: 😩

Comment (McRaeAlex) [reply tests/test_api.py:150]: Where do we get to determine the order of the tests?

Comment (McRaeAlex) [reply tests/test_api.py:150]: Also i would prefer to remove this optimization. Which by the emoji I think you agree with 🤷 

Is there a noticeable difference with / without it?

Comment (shrik450) [reply tests/test_api.py]: > Where do we get to determine the order of the tests?

It's top to bottom, in the order they are defined in the files.

> Also i would prefer to remove this optimization.

I removed it and refactored it to two different tests - the first testing `NotYetGeneratedException` is raised when the `get_generated_*` methods are called for the first time on a new file. I've made this change and moved the tests closer together.

**Review by McRaeAlex — APPROVED** [review]
I don't think its a good idea to rely on the backend generating the NotYetGenerated Exception but other than that looks good

**Review by garymoon — APPROVED** [review]

**Pull request merged**